src/overlays/actors/ovl_En_Bubble/z_en_bubble.c

```diff
@@ -23,45 +26,393 @@ const ActorInit En_Bubble_InitVars = {
     (ActorFunc)EnBubble_Update,
     (ActorFunc)EnBubble_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bubble/func_809CBC00.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bubble/func_809CBCBC.s")
+static ColliderJntSphItemInit sJntSphItemsInit[] = {
+    {
+        { 0x00, { 0x00000000, 0x00, 0x04 }, { 0xFFCFD753, 0x00, 0x00 }, 0x00, 0x01, 0x01 },
+        { 0x00, { { 0x0000, 0x0000, 0x0000 }, 16 }, 100 },
+    },
+    {
+        { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00002824, 0x00, 0x00 }, 0x00, 0x79, 0x00 },
+        { 0x00, { { 0x0000, 0x0000, 0x0000 }, 16 }, 100 },
+    },
+};
+
+static ColliderJntSphInit sJntSphInit = {
+    { COLTYPE_UNK6, 0x11, 0x09, 0x39, 0x10, COLSHAPE_JNTSPH },
+    2,
+    sJntSphItemsInit,
+};
+
+static CollisionCheckInfoInit2 sColChkInfoInit2 = {
+    0x01, 0x0002, 0x0019, 0x0019, 0xFF,
+};
+
+static Vec3f sEffectAccel = { 0.0f, -0.5f, 0.0f };
+
+static Color_RGBA8 sEffectPrimColor = { 255, 255, 255, 255 };
+
+static Color_RGBA8 sEffectEnvColor = { 150, 150, 150, 0 };
+
+void EnBubble_SetDimensions(EnBubble* this, f32 dim) {
+    f32 a;
+    f32 b;
+    f32 c;
+    f32 d;
+
+    this->actor.flags |= 1;
+    Actor_SetScale(&this->actor, 1.0f);
+    this->actor.shape.unk_08 = 16.0f;
+    this->graphicRotSpeed = 16.0f;
+    this->graphicEccentricity = 0.08f;
+    this->expansionWidth = dim;
+    this->expansionHeight = dim;
+    a = Rand_ZeroOne();
+    b = Rand_ZeroOne();
+    c = Rand_ZeroOne();
+    this->unk_218 = 1.0f;
+    this->unk_21C = 1.0f;
+    d = (a * a) + (b * b) + (c * c);
+    this->unk_1FC.x = a / d;
+    this->unk_1FC.y = b / d;
+    this->unk_1FC.z = c / d;
+}
+
+u32 func_809CBCBC(EnBubble* this) {
+    ColliderBody* body = &this->colliderSphere.list->body;
+
+    body->toucher.flags = 0x8;
+    body->toucher.effect = 0;
+    body->toucher.damage = 4;
+    body->toucherFlags = 1;
+    this->actor.velocity.y = 0.0f;
+    return 6;
+}
+
+// only called in an unused actionFunc
+u32 func_809CBCEC(EnBubble* this) {
+    EnBubble_SetDimensions(this, -1.0f);
+    return 12;
+}
+
+void EnBubble_DamagePlayer(EnBubble* this, GlobalContext* globalCtx) {
+    u32 damage = -this->colliderSphere.list->body.toucher.damage;
```

Comment:
Can this be `s32`? That would make a lot more sense since it's literally negative
```suggestion
    s32 damage = -this->colliderSphere.list->body.toucher.damage;
```